unstructured/embed/openai.py

```diff
@@ -0,0 +1,57 @@
+from typing import List, Optional
+
+import numpy as np
+
+from unstructured.documents.elements import (
+    Element,
+)
+from unstructured.embed.interfaces import BaseEmbeddingEncoder
+from unstructured.ingest.error import EmbeddingEncoderConnectionError
+from unstructured.utils import requires_dependencies
+
+
+class OpenAIEmbeddingEncoder(BaseEmbeddingEncoder):
+    def __init__(self, api_key: str, model_name: Optional[str] = "text-embedding-ada-002"):
+        self.api_key = api_key
+        self.model_name = model_name
+        self.initialize()
+
+    def initialize(self):
+        self.openai_client = self.get_openai_client()
+
+    def num_of_dimensions(self):
+        return np.shape(self.examplary_embedding)
```

Comment:
pretty sure every method that calls `self.examplary_embedding` would raise an error is called before that's being set. Should add some logic around that.

Reply:
self.examplary_embedding is set in the __init__ method, via this route:

__init__ -> initialize -> get_openai_client 

would that work?

Reply:
Perfect.